swap_meet/vendor.py

```diff
@@ -1,2 +1,82 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        if their_item not in vendor.inventory or my_item not in self.inventory:
+            return False
+        else:
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+            return True
+
+    def swap_first_item(self, vendor):
+        if len(self.inventory) == 0 or len(vendor.inventory) == 0:
+            return False
+        else:
+            my_new_item = vendor.inventory[0]
+            their_new_item = self.inventory[0]
+            self.swap_items(vendor, their_new_item, my_new_item)
+            return True
+
+    def get_best_by_category(self, category):
+        best_quality = 0
+        try:
+            for item in self.inventory:
+                if item.category == category:
+                    if item.condition > best_quality:
+                        best_quality = item.condition
+                        best_item = item
+            else:
+                return best_item 
+        except UnboundLocalError as err:
```

Comment:
We don't usually catch `UnboundLocalError`. That error tells us that we're accessing a variable that's defined in the function, but which because of the path through the function wasn't properly initialized. Consider here what happens if either the inventory is empty or there are no examples of the desired category. Line 49 will never be run, and the variable will never be initialized. Since we want the return value in this scenario to be `None`, we could add a line initializing `best_item` to `None` at line 44 and remove the try/except and the else of the for (using else with a loop is usually hard to understand). Instead, if we had a matching item, `best_item` would get updated in the loop, but if there wasn't , it would remain `None`. No matter what, when we return `best_item` at the end of the function, it would have the correct value.